projects/core/src/site-theme/store/actions/site-themes.action.ts

```diff
@@ -0,0 +1,48 @@
+/*
+ * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Action } from '@ngrx/store';
+import { SiteTheme } from '../../../model/misc.model';
+
+export const LOAD_SITE_THEMES = '[Site-theme] Load site themes';
+export const LOAD_SITE_THEMES_FAIL = '[Site-theme] Load site themes Fail';
+export const LOAD_SITE_THEMES_SUCCESS = '[Site-theme] Load site themes Success';
+export const SET_ACTIVE_SITE_THEME = '[Site-theme] Set Active site theme';
+export const SITE_THEME_CHANGE = '[Site-theme] site theme Change';
+
+export class LoadSiteThemes implements Action {
+  readonly type = LOAD_SITE_THEMES;
+}
+
+export class LoadSiteThemesFail implements Action {
+  readonly type = LOAD_SITE_THEMES_FAIL;
+  constructor(public payload: any) {}
+}
+
+export class LoadSiteThemesSuccess implements Action {
+  readonly type = LOAD_SITE_THEMES_SUCCESS;
+  constructor(public payload: SiteTheme[]) {}
+}
+
+export class SetActiveSiteTheme implements Action {
+  readonly type = SET_ACTIVE_SITE_THEME;
+  constructor(public payload: string) {}
+}
+
+export class SiteThemeChange implements Action {
```

Comment:
`query:` Can you help me to understand what's the purpose of this action?

Reply:
this is action for theme changes, activateSiteTheme$ is an effect that listens to changes in the site theme, buffers those changes, and then dispatches an action when both previous and current are not same.

Reply:
Ok, now I see that the reason for this might be tracking the whole flow - thanks to dispatching this action we can verify in the Redux DevTools that the theme changed (and that's all). Ok, let's keep it.